packages/core/src/utils/TypeUtil.ts

```diff
@@ -795,3 +810,11 @@ export interface WriteContractArgs {
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
   abi: any
 }
+
+export type ChainAdapter = {
```

Comment:
We should be moving this to `common`. Double check and remove here if necessary.

Reply:
For now, it's only used in `core` package, so let's keep here.